client/src/main/kotlin/tech/figure/objectstore/gateway/client/GatewayClient.kt

```diff
@@ -211,6 +215,178 @@ class GatewayClient(val config: ClientConfig) : Closeable {
+                request.scopeAddress = scopeAddress
+                request.granteeAddress = granteeAddress
+                grantId?.also { request.grantId = it }
+            }.build()
+        )
+        .get()
+
+    /**
+     * Revokes permission from the grantee to view the records associated with the given Provenance Blockchain Scope
+     * address.  The caller of this function has to be one of the following to avoid request rejection:
+     * - The associated scope's value owner
+     * - The master administrator of the gateway application
+     * - The grantee (accounts that have been granted permissions to a scope can revoke their own permissions if desired)
+     *
+     * @param scopeAddress The bech32 Provenance Blockchain Scope address for which to revoke access
+     * @param granteeAddress The bech32 Provenance Blockchain Account address for which access will be revoked
+     * @param keyPair the KeyPair of the key to sign the request with
+     * @param grantId A free-form grant identifier that will be used to query for existing scope_permissions records.
+     * If this value is omitted, all grants for the given scope and grantee combination will be revoked, versus targeting
+     * a singular unique record with the given id.
+     * @param timeout an optional timeout for the request
+     */
+    fun revokeScopePermission(
+        scopeAddress: String,
+        granteeAddress: String,
+        keyPair: KeyPair,
+        grantId: String? = null,
+        timeout: Duration = Duration.ofSeconds(10),
+    ): RevokeScopePermissionResponse = revokeScopePermission(
+        scopeAddress = scopeAddress,
+        granteeAddress = granteeAddress,
+        jwt = createJwt(keyPair, OffsetDateTime.now().plus(timeout)),
+        grantId = grantId,
+        timeout = timeout,
+    )
+
+    /**
+     * Revokes permission from the grantee to view the records associated with the given Provenance Blockchain Scope
+     * address.  The caller of this function has to be one of the following to avoid request rejection:
+     * - The associated scope's value owner
+     * - The master administrator of the gateway application
+     * - The grantee (accounts that have been granted permissions to a scope can revoke their own permissions if desired)
+     *
+     * @param scopeAddress The bech32 Provenance Blockchain Scope address for which to revoke access
+     * @param granteeAddress The bech32 Provenance Blockchain Account address for which access will be revoked
+     * @param keyRef the KeyRef of the key to sign the request with
+     * @param grantId A free-form grant identifier that will be used to query for existing scope_permissions records.
+     * If this value is omitted, all grants for the given scope and grantee combination will be revoked, versus targeting
+     * a singular unique record with the given id.
+     * @param timeout an optional timeout for the request
+     */
+    fun revokeScopePermission(
```

Comment:
I definitely see why you added in that new JWT scheme, it is getting ridiculous adding 3x every new client function 😆